src/Menu.php

```diff
@@ -32,6 +34,35 @@ public function add($title, $route)
         return $this->createItem($title, $route);
     }
 
+    /**
+     * Add a new menuItem to the menu if the condition is true.
+     *
+     * @param $title
+     * @param $route
+     * @param $condition
+     * @return bool|Contracts\MenuItem
+     */
+    public function addIf($title, $route, $condition)
+    {
+        return $this->resolveCondition($condition) ? $this->add($title, $route) : null;
+    }
+
+    /**
+     * Add a new menuItem to the menu when authorized.
+     *
+     * @param $title
+     * @param $route
+     * @param string|array $authorization
+     * @return null|Contracts\MenuItem
+     */
+    public function addIfCan($title, $route, $authorization)
+    {
+        $arguments = is_array($authorization) ? $authorization : [$authorization];
+        $ability = array_shift($arguments);
+
+        return $this->addIf(app(Gate::class)->allows($ability, $arguments), $title, $route);
```

Comment:
Usage of the `app()` helper requires laravel/foundation which requires the container again and a lot of other packages. Could you use the container directly instead in order to keep requirements to a minimum? And add the https://github.com/illuminate/container package to requirements in composer.json.

```php
Container::getInstance()->make(Gate::class)->allows($ability, $arguments)
```